src/Raven.Server/EventListener/GcEventsHandler.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.EventListener;
+
+public class GcEventsHandler : AbstractEventsHandler<GcEventsHandler.GCEventBase>
+{
+    private Dictionary<long, (DateTime DateTime, uint Generation, uint Reason)> _timeGcStartByIndex = new();
+    private EventWrittenEventArgs _suspendData;
+    private DateTime? _timeGcRestartStart;
+    private DateTime? _timeGcFinalizersStart;
+
+    protected override HashSet<EventType> DefaultEventTypes => EventListenerToLog.GcEvents;
+
+    public GcEventsHandler(Action<GCEventBase> onEvent, HashSet<EventType> eventTypes = null, long minimumDurationInMs = 0)
+    {
+        Update(eventTypes, minimumDurationInMs);
+        OnEvent = onEvent;
+    }
+
+    protected override Action<GCEventBase> OnEvent { get; }
+
+    public override bool HandleEvent(EventWrittenEventArgs eventData)
+    {
+        switch (eventData.EventName)
+        {
+            case "GCStart_V2":
```

Comment:
'm wondering if we should not assert all of them in that convention test in case one of them will change when framework will update, thoughts?

Reply:
done